solr/core/src/java/org/apache/solr/response/SolrQueryResponse.java

```diff
@@ -217,17 +219,15 @@ public String getToLogAsString() {
 
   /** Returns a string of the form "prefix name1=value1 name2=value2 ..." */
   public String getToLogAsString(String prefix) {
-    StringBuilder sb = new StringBuilder(prefix);
-    for (int i = 0; i < toLog.size(); i++) {
-      if (sb.length() > 0) {
-        sb.append(' ');
-      }
-      String name = toLog.getName(i);
-      Object val = toLog.getVal(i);
-      if (name != null) {
-        sb.append(name).append('=');
-      }
-      sb.append(val);
+    StringBuilder sb = new StringBuilder();
+    if (prefix != null && !prefix.isEmpty()) {
+      // TODO: remove "prefix"
+      Map<String, Object> keyValPairs = new LinkedHashMap<>();
+      keyValPairs.put("prefix", prefix);
+      keyValPairs.putAll(toLog);
+      sb.append(JSONUtil.toJSON(keyValPairs, -1));
+    } else {
+      sb.append(JSONUtil.toJSON(toLog, -1));
```

Comment:
Beautiful.  @gus-asf  with the move to JSON here, generating JSON is now a one-liner and not custom code.  Note the "prefix" thing ought to go away (separate issue).